### PR TITLE
[SQLRunDB] Fix abuse of sessions

### PR DESCRIPTION
**Pull request by alonmr**
Since the SQLRunDB is effectively a singleton (the factory holds its instance) we encountered the following errors:
1. We cannot use the session in multithreading until we move to async engine.
2. If a session query fails and not rolled back the session dies and can't be use anymore.
3. Keeping an open connection to the DB is occupying it and we see queries wait forever when the API is idle.

Here we create a new session for each operation (the session is not used from outside of the run db) and we make sure to close it when operation is done.
We still have other places where we create sessions in the API and we access the db layer (SQLDB) directly and we can improve this  mechanism but this should fix the issues we see for now.

### CODE REVIEW

File: mlrun/api/rundb/sqldb.py [hunk 1]
```diff
@@ -356,10 +334,16 @@ def delete_artifacts_tags(
...
 
     def store_schedule(self, data):
-        return self._transform_db_error(self.db.store_schedule, self.session, data)
+        session = create_session()
+        try:
+            return self._create_session_and_transform_db_error(
+                self.db.store_schedule, session, data
+            )
+        finally:
+            session.close()
```
Comment (quaark) [inline mlrun/api/rundb/sqldb.py:343]: leftover

File: mlrun/api/rundb/sqldb.py [hunk 2]
```diff
@@ -570,30 +534,32 @@ def patch_feature_set(
...
+        try:
+            return self._create_session_and_transform_db_error(
+                mlrun.api.crud.FeatureStore().create_feature_vector,
+                session,
+                project,
+                feature_vector,
+                versioned,
+            )
+        finally:
+            session.close()
```
Comment (quaark) [inline mlrun/api/rundb/sqldb.py:556]: leftover

**Review by quaark — CHANGES_REQUESTED** [review]
Looks good!
You've got a couple leftovers

**Review by quaark — APPROVED** [review]

**Comment by liranbg** [comment]
@george0st while I see your point, this is not a regression but just in-between release bug. beforehand, the object was used in a sync way. on 1.5.0, multiple threads uses it, thus needed

**Pull request merged**